### PR TITLE
Support running Octo from .net core tool-manifest

### PR DESCRIPTION
**Pull request by jeremyabbott**
### Description

Adds support for running `octo.exe` as a .NET Core tool installed via a tool-manifest.

Local verification (macOS):

```bash
dotnet new fake
dotnet tool install Octopus.DotNet.Cli
```

```fsharp
#r "paket:
source https://api.nuget.org/v3/index.json
source /Users/jeremy/code/FAKE/src/app/Fake.Tools.Octo/bin/Debug
nuget Fake.DotNet.Cli
nuget Fake.IO.FileSystem
nuget Fake.Core.Target
nuget Fake.Tools.Octo 1.0.0 //"
#load ".fake/build.fsx/intellisense.fsx"
open Fake.Core
open Fake.DotNet
open Fake.Core.TargetOperators

Target.initEnvironment ()

Target.create "Octo" (fun _ ->
    
    Fake.Tools.Octo.listEnvironments
      (fun opts ->
        { opts with 
            ToolType = ToolType.CreateLocalTool()
            Server = { opts.Server with ServerUrl = "https://octopus.server"; ApiKey="API-ABCDEFG123"}
        })
)

Target.runOrDefault "Octo"
```

## TODO

There are two obsolete calls in this module, but I'm not sure how to use the suggested replacement method.

- [x] New (API-)documentation for new features exist (Note: API-docs are enough, additional docs are in `help/markdown`)
- [x] unit or integration test exists (or short reasoning why it doesn't make sense)
  
- [ ] boy scout rule: "leave the code behind in a better state than you found it" (fix warnings, obsolete members or code-style in the places you worked in)     
- [x] (if new module) the module has been linked from the "Modules" menu, edit `help/templates/template.cshtml`, linking to the API-reference is fine.
- [x] (if new module) the module is in the correct namespace
- [x] (if new module) the module is added to Fake.sln (`dotnet sln Fake.sln add src/app/Fake.*/Fake.*.fsproj`)
- [x] Fake 5 [API guideline](https://fake.build/contributing.html#API-Design-Guidelines) is honored

### CODE REVIEW

**Comment by matthid** [comment]
Shouldn't we use a similar API as existing modules are using (https://fake.build/dotnet-cli.html#SDK-tools-local-global-clireference) for example [ReportGenerator](https://github.com/fsharp/FAKE/pull/2399/files#diff-6bd782ab06dfa727e4e35ce4bbaae43c)?

**Comment by jeremyabbott** [comment]
@matthid I'll take a look! Thank you for the suggestion.

**Comment by jeremyabbott** [comment]
@matthid I'm sorry for taking so long to come back to this! I made the recommended changes and tested them locally. I'm on macOS:

```bash
dotnet new fake
dotnet tool install Octopus.DotNet.Cli
```

```fsharp
#r "paket:
source https://api.nuget.org/v3/index.json
source /Users/jeremy/code/FAKE/src/app/Fake.Tools.Octo/bin/Debug
nuget Fake.DotNet.Cli
nuget Fake.IO.FileSystem
nuget Fake.Core.Target
nuget Fake.Tools.Octo 1.0.0 //"
#load ".fake/build.fsx/intellisense.fsx"
open Fake.Core
open Fake.DotNet
open Fake.Core.TargetOperators

Target.initEnvironment ()

Target.create "Octo" (fun _ ->
    
    Fake.Tools.Octo.listEnvironments
      (fun opts ->
        { opts with 
            ToolType = ToolType.CreateLocalTool()
            Server = { opts.Server with ServerUrl = "https://octopus.server"; ApiKey="API-ABCDEFG123"}
        })
)

Target.runOrDefault "Octo"
```

File: help/markdown/fake-tools-octo.md [hunk 1]
```diff
@@ -19,6 +19,13 @@ You will also need to install and configure at least one [Tentacle](http://octop
 
 This module is a wrapper around the [Octo.exe](https://octopus.com/docs/api-and-integration/octo.exe-command-line) CLI tool which controls Octopus Deploy API. You'll need the Octo.exe tool itself accessible to your FAKE script. Download it from [here](https://octopus.com/downloads).
 
+This module also supports use via a .NET Core Tool-Manifest. Installation is simple! From the root of your repository run the following
+
+```bash
+dotnet new tool-manifest # if one doesn't already exist
+dotnet tool install Octopus.DotNet.Cli 
+```
```
Comment (matthid) [inline help/markdown/fake-tools-octo.md:27]: Shouldn't we add here that you need to set the ToolPath and maybe link to https://fake.build/dotnet-cli.html#SDK-tools-local-global-clireference?

**Comment by matthid** [comment]
No need to be sorry, good things take time :)
Looks good to me in general, we could improve the docs and add some tests but that is nitpicking...

**Comment by jeremyabbott** [comment]
@matthid I'll follow up with those suggestions in a separate PR. Thank you again!

**Pull request merged**